### PR TITLE
3 training data generator numeric

### PR DESCRIPTION
**Pull request by iBoot32**
Changes:

- TrainingDataNumeric constructor
- TrainingDataNumeric `ReadParameterFile(...)`
- `GenerateMultivariateSamples(...)`
- `GenerateTrainingDataNumeric(...)`

Sampling from multivariate gaussian requires some linear algebra, so I include the `Eigen` library in `CMakeLists` now.
Also added `--rerun-on-failure` to ctest, not super important so can be removed in merge conflict

### CODE REVIEW

File: include/TrainingDataGeneratorNumeric.hpp [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+
+using Eigen::MatrixXd;
+using Eigen::VectorXd;
+
 class TrainingDataGeneratorNumeric
 {
 private:
-    /* data */
+    // Attributes
+    std::string _output_csv_file;
```
Comment (Coop8) [inline include/TrainingDataGeneratorNumeric.hpp:22]: Are variables supposed to be camel case?

**Review by Coop8 — COMMENTED** [review]
Everything here looks good. Only thing I would change is variables names to match our standard. Otherwise good work!

**Comment by iBoot32** [comment]
Updated all variable names to be camelCase

**Review by Coop8 — APPROVED** [review]
Very nice 😊

File: test/resources/param_numeric_out.txt
Comment (galaxybomb23) [inline test/resources/param_numeric_out.txt:1]: maybe we don't commit this

Comment (galaxybomb23) [inline test/resources/param_numeric_out.txt:1]: do we need this in our commit?

Comment (iBoot32) [reply test/resources/param_numeric_out.txt:1]: no

**Pull request merged**